packages/backend/src/app/database/database-connection.ts

```diff
@@ -46,3 +47,25 @@ export const databaseConnection =
   databaseType === DatabaseType.SQLITE3
       ? createSqlLiteDatasource()
       : createPostgresDataSource()
+
+export function APArrayContains<T extends Record<string, unknown>>(columnName: string, values: string[], query: SelectQueryBuilder<T>): SelectQueryBuilder<T> {
+    const databaseType = system.get(SystemProp.DB_TYPE)
+    switch (databaseType) {
+        case DatabaseType.POSTGRES:
+            return query.andWhere({
+                [columnName]: ArrayContains(values),
+            })
+        case DatabaseType.SQLITE3: {
+            const likeConditions = values.map((tag, index) => `flow_run.tags LIKE :tag${index}`).join(' AND ')
+            const likeParams = values.reduce((params, tag, index) => {
+                return {
+                    ...params,
+                    [`tag${index}`]: `%${tag}%`,
+                }
+            }, {})
+            return query.andWhere(likeConditions, likeParams)
+        }
+        default:
+            throw new Error(`Unsupported database type: ${databaseType}`)
+    }
+}
```

Comment:
wow! the decision to incorporate `sqlite` is making a lot of noise, we are basically doing everything db-related two times, I think we should either drop support for `sqlite` or if adoption numbers justify the use of `sqlite` let's see if we can switch to another ORM that can make things a little bit uniform across multiple DBs, right now TypeORM does little to nothing in this regard.